mesos/src/main/java/tachyon/mesos/TachyonFramework.java

```diff
@@ -0,0 +1,216 @@
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.mesos;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.mesos.Protos;
+import org.apache.mesos.Scheduler;
+import org.apache.mesos.SchedulerDriver;
+import org.apache.mesos.MesosSchedulerDriver;
+
+import tachyon.Constants;
+import tachyon.conf.TachyonConf;
+import tachyon.util.io.PathUtils;
+
+public class TachyonFramework {
+  static class TachyonScheduler implements Scheduler {
+    private boolean mMasterLaunched = false;
+
+    @Override
+    public void disconnected(SchedulerDriver driver) {
+      System.out.println("Disconnected from master.");
+    }
+
+    @Override
+    public void error(SchedulerDriver driver, String message) {
+      System.out.println("Error: " + message);
+    }
+
+    @Override
+    public void executorLost(SchedulerDriver driver, Protos.ExecutorID executorId,
+        Protos.SlaveID slaveId, int status) {
+      System.out.println("Executor " + executorId.getValue() + " was lost.");
+    }
+
+    @Override
+    public void frameworkMessage(SchedulerDriver driver, Protos.ExecutorID executorId,
+        Protos.SlaveID slaveId, byte[] data) {
+      System.out.println("Executor: " + executorId.getValue() + ", " + "Slave: "
+          + slaveId.getValue() + ", " + "Data: " + data + ".");
+    }
+
+    @Override
+    public void offerRescinded(SchedulerDriver driver, Protos.OfferID offerId) {
+      System.out.println("Offered " + offerId.getValue() + " rescinded.");
+    }
+
+    @Override
+    public void registered(SchedulerDriver driver, Protos.FrameworkID frameworkId,
+        Protos.MasterInfo masterInfo) {
+      System.out.println("Registered framework " + frameworkId.getValue() + " with master "
+          + masterInfo.getHostname() + ":" + masterInfo.getPort() + ".");
+    }
+
+    @Override
+    public void reregistered(SchedulerDriver driver, Protos.MasterInfo masterInfo) {
+      System.out.println("Registered framework with master " + masterInfo.getHostname() + ":"
+          + masterInfo.getPort() + ".");
+    }
+
+    @Override
+    public void resourceOffers(SchedulerDriver driver, List<Protos.Offer> offers) {
+      final double MASTER_CPUS = 1;
+      final double MASTER_MEM = 128;
```

Comment:
What is the unit for `MASTER_MEM`? `MB`?  Also, would it be reasonable to make these two values configurable?